### ISSUE TITLE
`realpath` typing

### ISSUE THREAD

**Issue by zerkms** [vimeo/psalm#3371]
```php
<?php
declare(strict_types=1);

class Foo {
    private string $path;
    
    public function __construct(string $path) {
        $this->path = realpath($path);
    }
}

var_dump(new Foo('q'));
```

https://psalm.dev/r/4a379826cc

`realpath` return type (if used independently) is a correct `false|string`, but when used in code like that - no type error is triggered.

**Comment by weirdan** [comment]
You need to enable `Use strict internal function results` on psalm.dev: https://psalm.dev/r/685cf0d4ef

**Comment by weirdan** [comment]
It translates to `ignoreInternalFunctionFalseReturn=false` + `ignoreInternalFunctionNullReturn=false` (both are false by default) in regular `psalm.xml` config. See https://psalm.dev/docs/running_psalm/configuration/#ignoreinternalfunctionfalsereturn
